openqabot/loader/gitea.py

```diff
@@ -16,78 +19,78 @@
 from lxml import etree
 from osc.core import MultibuildFlavorResolver
 
-from .. import GIT_REVIEW_BOT, GITEA, OBS_DOWNLOAD_URL, OBS_GROUP, OBS_PRODUCTS, OBS_REPO_TYPE, OBS_URL
-from ..types import Repos
-from ..utils import retry10 as retried_requests
+from openqabot import GIT_REVIEW_BOT, GITEA, OBS_DOWNLOAD_URL, OBS_GROUP, OBS_PRODUCTS, OBS_REPO_TYPE, OBS_URL
+from openqabot.utils import retry10 as retried_requests
+
+if TYPE_CHECKING:
+    from openqabot.types import Repos
 
 log = getLogger("bot.loader.gitea")
 
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
 
-def make_token_header(token: str) -> Dict[str, str]:
+def make_token_header(token: str) -> dict[str, str]:
     return {} if token is None else {"Authorization": "token " + token}
 
 
-def get_json(query: str, token: Dict[str, str], host: str = GITEA) -> Any:
+def get_json(query: str, token: dict[str, str], host: str = GITEA) -> Any:
     return retried_requests.get(host + "/api/v1/" + query, verify=False, headers=token).json()
 
 
-def post_json(query: str, token: Dict[str, str], post_data: Any, host: str = GITEA) -> Any:
+def post_json(query: str, token: dict[str, str], post_data: Any, host: str = GITEA) -> Any:
     url = host + "/api/v1/" + query
     res = retried_requests.post(url, verify=False, headers=token, json=post_data)
     if not res.ok:
         log.error("Unable to POST %s: %s", url, res.text)
 
 
 def read_utf8(name: str) -> str:
-    with open("responses/%s" % name, "r", encoding="utf8") as utf8:
-        return utf8.read()
+    return Path("responses/{}".format(name)).read_text(encoding="utf8")
 
 
 def read_json(name: str) -> Any:
-    with open("responses/%s.json" % name, "r", encoding="utf8") as json_file:
-        return json.loads(json_file.read())
+    return json.loads(Path("responses/{}.json".format(name)).read_text(encoding="utf8"))
 
 
 def read_xml(name: str) -> etree.ElementTree:
-    return etree.parse("responses/%s.xml" % name)
+    return etree.parse("responses/{}.xml".format(name))
 
 
 def reviews_url(repo_name: str, number: int) -> str:
-    # https://docs.gitea.com/api/1.20/#tag/repository/operation/repoListPullReviews
-    return "repos/%s/pulls/%s/reviews" % (repo_name, number)
+    # https://docs.gitea.com/api/1.20/#tag/repository/operation/repolistPullReviews
+    return "repos/{}/pulls/{}/reviews".format(repo_name, number)
 
 
 def changed_files_url(repo_name: str, number: int) -> str:
     # https://docs.gitea.com/api/1.20/#tag/repository/operation/repoGetPullRequestFiles
-    return "repos/%s/pulls/%s/files" % (repo_name, number)
+    return "repos/{}/pulls/{}/files".format(repo_name, number)
 
 
 def comments_url(repo_name: str, number: int) -> str:
     # https://docs.gitea.com/api/1.20/#tag/issue/operation/issueCreateComment
-    return "repos/%s/issues/%s/comments" % (repo_name, number)
+    return "repos/{}/issues/{}/comments".format(repo_name, number)
```

Comment:
nitpick: Can we have fstring for those changes? somewhere below you do use fstring. I would say to adapt one of the two. ofc I am in favor of the fstring.